salt/_states/metalk8s_kubernetes.py

```diff
@@ -1505,3 +1505,79 @@ def customresourcedefinition_present(
             'new': res}
 
     return ret
+
+
+def node_cordon(
+        name,
+        **kwargs):
+    ret = {'name': name,
+           'changes': {},
+           'result': False,
+           'comment': ''}
+
+    unschedulable = __salt__[
+        'metalk8s_kubernetes.node_unschedulable'](name, **kwargs)
+
+    if unschedulable is None:
+        ret['comment'] = 'Unable to get node {0}'.format(name)
+        return ret
+
+    if unschedulable is True:
+        ret['result'] = True
+        ret['comment'] = 'The node {0} is already cordoned'.format(name)
+        return ret
+
+    if __opts__['test']:
+        ret['comment'] = 'The node {0} is going to be cordoned'.format(name)
+        ret['result'] = None
+        return ret
+
+    res = __salt__['metalk8s_kubernetes.node_cordon'](node_name=name, **kwargs)
+
+    ret['result'] = True
+    ret['changes'] = {
+        'old': {'unschedulable': unschedulable},
+        'new': {'unschedulable': res['spec']['unschedulable']}
+    }
+    ret['comment'] = 'Node {0} cordoned'.format(name)
+
+    return ret
+
+
+def node_uncordon(
```

Comment:
Same comments as with `node_cordon`.

Also, these functions are basically symmetrical, so would it make sense to factor this out, have one 'helper' (private) state which can then be invoked with a boolean that signals the 'intended' `unschedulable` value?